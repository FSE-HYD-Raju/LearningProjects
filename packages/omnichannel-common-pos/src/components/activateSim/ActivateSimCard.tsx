import * as React from "react";
import {
	ActivateSimConfigurationType,
	ActivateSimOrderData,
	Order,
	OrderItem,
	User,
	SimIccVerification,
	SimIccVerificationAttributes
} from "../../redux/types";
import ActivateSimBanner from "./ActivateSimBanner";
import OrderUtil from "../../utils/order/OrderUtil";

export interface ActivateSimCardStateProps {
	activateSimOrderData?: ActivateSimOrderData;
	user?: User;
	configuration: ActivateSimConfigurationType;
	verificationResponse?: SimIccVerification;
	verificationError: boolean;
}

export interface ActivateSimCardActionProps {
	actions: {
		getInProgressOrders: (personId: string) => void;
		simIccVerification: (payload: SimIccVerificationAttributes) => void;
		onClose: () => void;
	};
}

export type ActivateSimCardProps = ActivateSimCardStateProps & ActivateSimCardActionProps;

export type SimCardActivationInfo = {
	iccid: string;
	orderReference: string;
	tocTocUrl: string;
	iccidVerification: boolean;
};

class ActivateSimCard extends React.PureComponent<ActivateSimCardProps> {
	constructor(props: ActivateSimCardProps) {
		super(props);
	}

	componentWillMount() {
		const { user, actions } = this.props;
		if (user && user.id) {
			actions.getInProgressOrders(user.id);
		}
	}

	getSimCardActivationInfo = (): SimCardActivationInfo[] => {
		const { activateSimOrderData, configuration } = this.props;
		const orders = activateSimOrderData && activateSimOrderData.orders;

		const result: SimCardActivationInfo[] = [];

		if (this.isConfigurationValid() && orders && orders.length) {
			orders.forEach((order: Order) => {
				const orderDetails = order.attributes && order.attributes.orderDetails;
				const orderReference = order.attributes && order.attributes.referenceNumber;

				const iccid =
					orderDetails &&
					orderDetails.msisdnIccid &&
					orderDetails.msisdnIccid.length &&
					orderDetails.msisdnIccid[0].iccid;

				const token = orderDetails && orderDetails.token;

				if (iccid && orderReference) {
					const allOrderItems = OrderUtil.getAllOrderItems(order);

					const activationItem = allOrderItems.find(item =>
						this.isItemContainsOneOfPOs(item, configuration.activatePOIds as string[])
					);

					const deliveryItem = allOrderItems.find(item =>
						this.isItemContainsOneOfPOs(item, configuration.deliveryPOIds as string[])
					);

					if (activationItem && deliveryItem) {
						const vcn = configuration.verificationCharacteristicName as string;
						const iccidVv = configuration.iccidVerificationValue as string;
						const videoVv = configuration.videoVerificationValue as string;

						const iccidVerification = this.isItemContainsCharacteristicWithValue(
							deliveryItem,
							vcn,
							iccidVv
						);
						const videoVerification = this.isItemContainsCharacteristicWithValue(
							deliveryItem,
							vcn,
							videoVv
						);

						if (iccidVerification || videoVerification) {
							result.push({
								iccid,
								orderReference,
								iccidVerification,
								tocTocUrl: `${token}`
							});
						}
					}
				}
			});
		}

		return result;
	};

	isConfigurationValid = () => {
		const { configuration } = this.props;

		return (
			configuration &&
			configuration.activatePOIds &&
			configuration.activatePOIds.length &&
			configuration.deliveryPOIds &&
			configuration.deliveryPOIds.length &&
			configuration.verificationCharacteristicName &&
			configuration.iccidVerificationValue &&
			configuration.videoVerificationValue
		);
	};

	isItemContainsOneOfPOs = (orderItem: OrderItem, pos: string[]): boolean => {
		const po = (orderItem.attributes && orderItem.attributes.productOffering) || orderItem.productOffering;

		return Boolean(po && pos.includes(po.id));
	};

	isItemContainsCharacteristicWithValue = (
		orderItem: OrderItem,
		characteristicName: string,
		characteristicValue: string
	): boolean => {
		const characteristics =
			(orderItem.attributes && orderItem.attributes.inputtedCharacteristics) || orderItem.inputtedCharacteristics;

		return Boolean(characteristics && characteristics[characteristicName] === characteristicValue);
	};

	render() {
		const { user } = this.props;

		const simCardActivationInfo = this.getSimCardActivationInfo();

		if (user && user.attributes && simCardActivationInfo.length) {
			const name = user.attributes.firstName;

			return (
				<ActivateSimBanner
					name={name}
					simCardActivationInfo={simCardActivationInfo}
					verificationResponse={this.props.verificationResponse}
					verificationError={this.props.verificationError}
					simIccVerification={this.props.actions.simIccVerification}
					onClose={this.props.actions.onClose}
				/>
			);
		} else {
			return null;
		}
	}
}

export default ActivateSimCard;
