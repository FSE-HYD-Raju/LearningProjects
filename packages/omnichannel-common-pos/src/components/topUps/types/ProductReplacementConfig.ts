interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
