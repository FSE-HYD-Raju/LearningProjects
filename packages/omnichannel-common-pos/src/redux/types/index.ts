import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
