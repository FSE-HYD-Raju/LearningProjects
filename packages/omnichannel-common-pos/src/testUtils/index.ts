import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
