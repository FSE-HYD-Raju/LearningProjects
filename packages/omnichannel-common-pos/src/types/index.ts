import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
