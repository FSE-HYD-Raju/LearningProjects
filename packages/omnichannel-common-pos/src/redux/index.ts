"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
