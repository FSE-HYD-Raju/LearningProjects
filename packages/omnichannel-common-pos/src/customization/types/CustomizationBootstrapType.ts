import { ConnectFluxReduxChannelType, FluxAlt, StoreConfig, AppState } from "../../redux";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";
import { Store } from "react-redux";

export type GetStoreConfig<S extends AppState> = {
	reducers?: object|undefined;
	customSaga?: StoreConfig<S>["customSaga"]|null;
	interceptReducers?: StoreConfig<S>["interceptReducers"]|undefined;
	history?: StoreConfig<S>["history"]|undefined;
	initialState?: StoreConfig<S>["initialState"]|undefined;
};

export interface CustomizationBootstrapType<S extends AppState> {
	getFlux: <S>(fluxCustomization: {
		customizedStores?: Record<string, any>,
		customizedActions?: Record<string, any>,
	}) => FluxAlt<S & AppState>;
	getStore: <S>(config: GetStoreConfig<S & AppState>) => Store<S>;
	connectFluxReduxChannel: ConnectFluxReduxChannelType;
	customizationPointsMapping: CustomizationPointsMapping<S>;
}
