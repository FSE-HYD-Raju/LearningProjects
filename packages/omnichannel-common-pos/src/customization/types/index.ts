import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
