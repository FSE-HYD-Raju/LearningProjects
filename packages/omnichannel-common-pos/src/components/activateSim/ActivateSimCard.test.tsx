import * as React from "react";
import { mountWithContext, shallowWithContext } from "../../testUtils";
import ActivateSimCard, { ActivateSimCardProps, SimCardActivationInfo } from "./ActivateSimCard";
import { ActivateSimOrderData, User, ActivateSimConfigurationType } from "../../redux/types";

const activateSimOrderData: ActivateSimOrderData = require("./testdata/activateSimOrdersData.json");

const user: User = {
	id: "dummy-user-id",
	attributes: {
		firstName: "Thomas"
	}
} as User;

const configuration: ActivateSimConfigurationType = {
	verificationCharacteristicName: "CH_Allowed_Doc_Verif_Method",
	iccidVerificationValue: "sim_iccid",
	videoVerificationValue: "video",
	activatePOIds: ["PO_change_sim"],
	deliveryPOIds: ["PO_HomeDelivery"]
};

describe("ActivateSimCard", () => {
	let props: ActivateSimCardProps;

	beforeEach(() => {
		props = {} as ActivateSimCardProps;
	});

	it("succeeds at shallow mount without props", () => {
		const wrapper = shallowWithContext(<ActivateSimCard {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount without props", () => {
		mountWithContext(<ActivateSimCard {...props} />);
	});

	it("should call getInProgressOrders on mount", () => {
		const getInProgressOrders = jest.fn();
		props = ({
			user,
			actions: {
				getInProgressOrders
			}
		} as any) as ActivateSimCardProps;

		mountWithContext(<ActivateSimCard {...props} />);

		expect(getInProgressOrders).toBeCalledWith(user.id);
	});

	it("should show banner for proper orders", () => {
		props = ({
			user,
			activateSimOrderData,
			configuration,
			actions: {
				getInProgressOrders: () => {}
			}
		} as any) as ActivateSimCardProps;

		const wrapper = mountWithContext(<ActivateSimCard {...props} />);

		const banner = wrapper.find("ActivateSimBanner");
		expect(banner.length).toBe(1);
		expect(banner.props().name).toBe("Thomas");

		const simCardActivationInfo: SimCardActivationInfo[] = banner.props().simCardActivationInfo;
		expect(simCardActivationInfo.length).toBe(1);

		expect(simCardActivationInfo[0].iccid).toBe("8939883295001002529");
		expect(simCardActivationInfo[0].orderReference).toBe("CH231020181631");
		expect(simCardActivationInfo[0].tocTocUrl).toBe(
			"http://toctoc.com?token=eyJhbGciOiJIUzI1NiJ9.eyJPcmRlcklkIjoiSGFyZEJfSG9tZURlbGl2ZXJ5XzE1MzgzODM2MzkifQ.q3DNQrYZpAVufkDU6CwCDhqnOe4f8n6v3-JEjwxq3zA"
		);
		expect(simCardActivationInfo[0].iccidVerification).toBeTruthy();
	});
});
