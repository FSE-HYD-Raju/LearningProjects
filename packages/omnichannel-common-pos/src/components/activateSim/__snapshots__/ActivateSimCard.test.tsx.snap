// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivateSimCard succeeds at shallow mount without props 1`] = `""`;
